### PR TITLE
Support Complex[float32] and Complex[float64]

### PR DESCRIPTION
**Pull request by Paalon**
From https://github.com/nim-lang/Nim/pull/7924 and https://github.com/nim-lang/Nim/pull/4666.
In this PR, I defined `Real = SomeInteger|SomeFloat` and `Number = Real|Complex`, I don't know it's the best name, but we need something type class for represent them. Some language support Complex[int] but here only floating point number is supported.

### CODE REVIEW

File: lib/pure/complex.nim [hunk 1]
```diff
@@ -22,147 +22,281 @@ const
...
-  result.re = x
-  result.im = 0
 
-proc `==` *(x, y: Complex): bool =
-  ## Compare two complex numbers `x` and `y` for equality.
-  result = x.re == y.re and x.im == y.im
+proc toComplex*[T: Number](x: T): auto =
+  ## Convert ``x`` to Complex.
+  when T is Complex:
+    var z: T
```
Comment (Araq) [inline lib/pure/complex.nim]: Nonsense to use a temporary ``z`` here.

File: lib/pure/complex.nim [hunk 2]
```diff
@@ -22,147 +22,281 @@ const
...
+    z = x
+    result = z
+  elif T is SomeInteger:
+    var z: Complex[float]
```
Comment (Araq) [inline lib/pure/complex.nim]: Convoluted.

File: lib/pure/complex.nim [hunk 3]
```diff
@@ -22,147 +22,281 @@ const
...
+    z.re = float(x)
+    z.im =  0.0
+    result = z
+  elif T is float32:
+    var z: Complex[float32]
```
Comment (Araq) [inline lib/pure/complex.nim]: Convoluted.

**Comment by krux02** [comment]
``Complex[float32]`` is now implemented.

**Pull request closed without merging**